pypesto/hierarchical/util.py

```diff
@@ -271,6 +285,149 @@ def apply_sigma(
         sigma[i][mask[i]] = sigma_value
 
 
+def compute_bounded_optimal_scaling_offset_coupled(
+    data: List[np.ndarray],
+    sim: List[np.ndarray],
+    sigma: List[np.ndarray],
+    s: InnerParameter,
+    b: InnerParameter,
+    s_opt_value: float,
+    b_opt_value: float,
+):
+    """Compute optimal scaling and offset of a constrained optimization problem.
+
+    Computes the optimal scaling and offset of a constrained optimization in
+    case the unconstrained optimization yields a value outside the bounds.
+    We know the optimal solution then lies on the boundary of the bounds.
+    If both parameters are unsatisfied, we need to check 2 unconstrained
+    problems, and evaluate at 3 edge points. Otherwise, if only one parameter
+    is unsatisfied, we need to check 1 unconstrained problem, and evaluate at
+    2 edge points. We then return the candidate point with the lowest objective
```

Comment:
Might be more understandable for some, if you replace some of the docstring with this:

In the 2D offset-scaling bounded (rectangular) space, after unconstrained optimization, if only one parameter is outside the bounds, then there is one active edge (constraint) of the rectangle. We perform optimization on this edge that is unconstrained in the other parameter. If this new optimum is outside the bounds of the other parameter, the nearest vertex is chosen as the optimum.

If both parameters are outside the bounds, then there are two active edges, which are optimized independently as above, then compared.